### PR TITLE
ehci: Fix multiple issues regarding controller stack and HID 

### PR DESCRIPTION
**Pull request by borrrden**
- The assertions happening in the EHCI controller regarding clean state transition appear to not be reliable for more than one device being connected to the controller.  The port connections will be happening simultaneously and there is no guaranteed starting state for them to be in.
- Implemented TODO for handing off port to UHCI in the event that it is not a high speed device
- Added the FEATURE (0xB0) token to the list of recognized tokens for HID

### CODE REVIEW

**Review by Dennisbonke — APPROVED** [review]
It looks good to me but needs another review.

**Comment by Dennisbonke** [comment]
@avdgrinten or @qookei can we review this PR?

**Comment by ArsenArsen** [comment]
please split up this commit and reword it to align with the rest of the repository

File: drivers/usb/hcds/ehci/src/ehci.hpp [hunk 1]
```diff
@@ -27,16 +27,14 @@ struct Enumerator {
 	// Called by the USB hub driver once a device completes reset.
 	void enablePort(int port);
 
-private:
-	enum class State {
-		null, resetting, probing
-	};
+	// Called by the USB hub driver if a port fails to enable after connection
+	void portDisabled(int port);
```
Comment (ArsenArsen) [inline drivers/usb/hcds/ehci/src/ehci.hpp:31]: This function could be named better

File: drivers/usb/devices/hid/src/main.cpp [hunk 2]
```diff
@@ -412,6 +412,11 @@ void HidDevice::parseReportDescriptor(Device, uint8_t *p, uint8_t* limit) {
 			local = LocalState();
 			break;
 
+		case 0xB0:
+			if(logDescriptorParser)
+				printf("usb-hid:      Feature: 0x%x\n", data);
```
Comment (ArsenArsen) [inline drivers/usb/devices/hid/src/main.cpp:417]: there's an extra space here in the string

**Review by ArsenArsen — CHANGES_REQUESTED** [review]

**Comment by no92** [comment]
Superseded by #545.
